elixir/apps/domain/lib/domain/auth/adapters/google_workspace/api_client.ex

```diff
@@ -173,16 +174,31 @@ defmodule Domain.Auth.Adapters.GoogleWorkspace.APIClient do
   defp list(uri, api_token, key) do
     request = Finch.build(:get, uri, [{"Authorization", "Bearer #{api_token}"}])
 
-    with {:ok, %Finch.Response{body: response, status: status}} when status in 200..299 <-
+    with {:ok, %Finch.Response{body: response, status: 200}} <-
            Finch.request(request, @pool_name),
          {:ok, json_response} <- Jason.decode(response),
-         {:ok, list} <- Map.fetch(json_response, key) do
+         {:ok, list} when is_list(list) <- Map.fetch(json_response, key) do
       {:ok, list, json_response["nextPageToken"]}
     else
-      {:ok, %Finch.Response{status: status}} when status in 500..599 ->
+      {:ok, %Finch.Response{status: status} = response} when status in 201..299 ->
+        Logger.warning("API request succeeded with unexpected 2xx status #{status}",
+          response: inspect(response)
+        )
+
         {:error, :retry_later}
 
-      {:ok, %Finch.Response{body: response, status: status}} ->
+      {:ok, %Finch.Response{status: status} = response} when status in 300..399 ->
+        Logger.warning("API request succeeded with unexpected 3xx status #{status}",
+          response: inspect(response)
+        )
+
+        {:error, :retry_later}
+
+      {:ok, %Finch.Response{body: response, status: status}} when status in 400..499 ->
+        Logger.error("API request failed with 4xx status #{status}",
+          response: inspect(response)
+        )
+
         case Jason.decode(response) do
```

Comment:
Not something that needs to be done in this PR, but I'm guessing we probably don't need this any more since we're logging the responses in all non `200` responses now.

Reply:
Hm, I think this might be returned so the caller can inject it into the email alert. Let me check.

Reply:
In that case, we've technically changed the logic here a bit, because previously `3xx` and `4xx` responses where caught by this, but now only `4xx` responses are caught.  In practice this may not be an issue, just wanted to make sure we're aware of the change.

Reply:
👍 Good point. I think if anything, we probably want to stick to this behavior right? It does change, but if we were recording errors for 3xx responses then that would probably have been a bug.

The thinking here was to err on the side of "Log and no-op" if we receive something unexpected.

Reply:
Yeah, for sure.  I don't think this will actually be an issue.